src/config.rs

```diff
@@ -1513,14 +1513,22 @@ impl Config {
 
     #[cfg(windows)]
     pub fn installed_config<'a>() -> Option<Cow<'a, Path>> {
-        if let Some(mut path) = ::std::env::home_dir() {
-            path.push("alacritty");
-            path.set_extension("yml");
-            if path.exists() {
-                return Some(path.into());
-            }
+        #[allow(deprecated)]
+        let old = ::std::env::home_dir()
+            .map(|path| path.join("alacritty.yml"));
+        let new = dirs::config_dir()
+            .map(|path| path.join("alacritty\\alacritty.yml"));
+        // TODO: Remove old configuration location warning (Deprecated 03/12/2018)
+        if let Some(old) = old.as_ref().filter(|old| old.exists()) {
+            warn!(
+                "Found old configuration at: '{}'. The file should be moved to the new location: '{}'.", 
+                old.to_string_lossy(), 
+                new.as_ref().map(|new| new.to_string_lossy()).unwrap_or_else(|| Cow::from("(UNKNOWN)"))
```

Comment:
I think an `expect(...)` here makes more sense.

Reply:
Do you mean expect on `dirs::config_dir` and `std::env::home_dir`? Those are the failable part of this.

Reply:
I was specifically talking about the `unwrap_or_else(|| Cow::from("(UNKNOWN)"))`, since that could actually lead to unexpected behavior. If there's no good way to recover, Alacritty should crash with a helpful error message rather than trying to fill in the gaps with garbage.

Reply:
It's the equivalent of failing if `~` doesn't exist on linux, so I think it's probably reasonable to panic.